internal/protoc/downloader.go

```diff
@@ -64,7 +71,25 @@ func newDownloader(config settings.Config, options ...DownloaderOption) *downloa
 	if downloader.config.Compile.ProtobufVersion == "" {
 		downloader.config.Compile.ProtobufVersion = vars.DefaultProtocVersion
 	}
-	return downloader
+	if downloader.protocBinPath != "" || downloader.protocWktPath != "" {
+		if downloader.protocURL != "" {
+			return nil, fmt.Errorf("cannot use protoc-url in combination with either protoc-bin-path or protoc-wkt-path")
+		}
+		if downloader.protocBinPath == "" || downloader.protocWktPath == "" {
+			return nil, fmt.Errorf("both protoc-bin-path and protoc-wkt-path must be set")
+		}
+		cleanBinPath := filepath.Clean(downloader.protocBinPath)
```

Comment:
Might want to check that this is executable, even better would be to make sure the version flag works as we check below

Reply:
We can leverage the current output and don't need to add this into the logic here:
```
fork/exec /path/to/bin: permission denied
```

Reply:
I guess if you want, I don’t think this is great error message and it’s a really easy check against the os.FileInfo, aka it’s what we effectively do in the Downloader if you don’t use this flag, so nice to be consistent